### DISCUSSION TITLE
Regarding the limit of embed fields to 25

### DISCUSSION THREAD

**Discussion by coool9** [bwmarrin/discordgo#1493] [Ideas]
I was scratching my head over it. Then read about it in the discord changelog. My idea is to add some kinda check that embed fields do not exceed 25

**Comment by MagnusHJensen** [comment]
@coool9 
When calling `ChannelMessageSendComplex` with an embed with more than 25 fields, it just returns Discords own validation error, see the error below. 

```json
{
    "message": "Invalid Form Body",
    "code": 50035,
    "errors": {
        "embeds": {
            "0": {
                "fields": {
                    "_errors": [
                        {
                            "code": "BASE_TYPE_MAX_LENGTH",
                            "message": "Must be 25 or fewer in length."
                        }
                    ]
                }
            }
        }
    }
}
```

TBH I think this is pretty clear, and it seems the library is heavily relying on Discord's own validation, to avoid building stale validation on top. It could save an API call to verify the length of 25 fields, but let's say Discord changes it to 40 tomorrow, that does not require a new version to be published, but bot developers can just use 40 fields the same day.
